src/main/kotlin/interpreter/stack/UserCodeStackFrame.kt

```diff
@@ -80,6 +97,31 @@ class UserCodeStackFrame(val function: PyUserFunction) : StackFrame() {
      */
     val lineNo: Int get() = function.code.getLineNumber(bytecodePointer)
 
+    /**
+     * Utility function for calling magic methods
+     */
+    fun magicMethod(obj: PyObject, magicName: String, param: PyObject?) {
+        val fn = obj.pyGetAttribute(magicName)
+        if (fn !is PyCallable) {
+            Exceptions.TYPE_ERROR("'${obj.type.name}'.$magicName is not callable.").throwKy()
+        }
+        val result = if (param != null) fn.runCallable(listOf(param)) else fn.runCallable(listOf())
+        stack.push(result)
+    }
+
+    fun magicMethod(obj: PyObject, magicName: String) = magicMethod(obj, magicName, null)
```

Comment:
You don't need this. Do `param: PyObject = null` and annotate with `@JvmOverloads`.

Reply:
Or just make PyObject nullable?